Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs

```diff
@@ -89,6 +90,20 @@ public override void ViewDidDisappear(bool animated)
 
 			_appeared = false;
 			PageController.SendDisappearing();
+
+			var navigationPage = Element.Parent as NavigationPage;
+			if (navigationPage == null)
+				return;
+
+			int managedStackCount = navigationPage.Navigation.NavigationStack.Count;
+			IVisualElementRenderer renderer = Platform.GetRenderer(navigationPage);
+
+			var navigationController = renderer as UINavigationController;
+			if (navigationController != null && managedStackCount > navigationController.ViewControllers.Length)
+			{
+				Task<Page> task = ((INavigationPageController)renderer.Element).PopAsyncInner(animated, true);
```

Comment:
Well, this is still better than a broken system. I'll try to find a solution on `NavigationRenderer` tonight, but I'd consider including this for now instead of potentially waiting for months to fix the navigation stack.